### DISCUSSION TITLE
[Proposal] Property extension

### DISCUSSION THREAD

**Discussion by JustNrik** [dotnet/csharplang#2772] [Language Ideas]
I don't know if this has been suggested before but we should be allowed to create property extensions. Example:

```cs
public static bool IsEven[this int value]
    => value % 2 == 0;
```

This may be useful to add properties to objects instead of creating methods (well, properties are methods under the hood)

**Comment by HaloFour** [comment]
IIRC the reason extension properties don't currently exist in the language is that this syntax was found incredibly unpalatable.  This is particularly true when the extension target is generic.

Check out #192 and https://github.com/dotnet/roslyn/issues/11159 for some syntax that's been suggested by the LDM to allow adding extension members to other classes.
